client/modules/IDE/components/Searchbar/Searchbar.jsx

```diff
@@ -1,82 +1,78 @@
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import React from 'react';
 import { throttle } from 'lodash';
 import { withTranslation } from 'react-i18next';
 import i18next from 'i18next';
 import SearchIcon from '../../../../images/magnifyingglass.svg';
 
-class Searchbar extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      searchValue: this.props.searchTerm
-    };
-    this.throttledSearchChange = throttle(this.searchChange, 500);
-  }
+const Searchbar = ({
+  searchTerm,
+  setSearchTerm,
+  resetSearchTerm,
+  searchLabel,
+  t
+}) => {
+  const [searchValue, setSearchValue] = useState(searchTerm);
 
-  componentWillUnmount() {
-    this.props.resetSearchTerm();
-  }
+  const searchChange = () => {
+    setSearchTerm(searchValue.trim());
+  };
+
+  const throttledSearchChange = throttle(searchChange, 500);
+
+  useEffect(
+    () => () => {
+      resetSearchTerm();
+    },
+    [resetSearchTerm]
+  );
+
+  const handleResetSearch = () => {
+    setSearchValue('');
+    resetSearchTerm();
+  };
 
-  handleResetSearch = () => {
-    this.setState({ searchValue: '' }, () => {
-      this.props.resetSearchTerm();
-    });
+  const handleSearchChange = (e) => {
+    const newValue = e.target.value;
+    setSearchValue(newValue);
+    throttledSearchChange(newValue.trim());
   };
 
-  searchChange = () => {
-    this.props.setSearchTerm(this.state.searchValue.trim());
+  Searchbar.propTypes = {
+    searchTerm: PropTypes.string.isRequired,
+    setSearchTerm: PropTypes.func.isRequired,
+    resetSearchTerm: PropTypes.func.isRequired,
+    searchLabel: PropTypes.string,
+    t: PropTypes.func.isRequired
   };
 
-  handleSearchChange = (e) => {
-    this.setState({ searchValue: e.target.value }, () => {
-      this.throttledSearchChange(this.state.searchValue.trim());
-    });
+  Searchbar.defaultProps = {
+    searchLabel: i18next.t('Searchbar.SearchSketch')
```

Comment:
Please keep the `propTypes` and `defaultProps` outside of the body of the function.

Reply:
sure